packages/db/src/lib/granule.ts

```diff
@@ -45,3 +45,31 @@ export const upsertGranuleWithExecutionJoinRecord = async (
   }
   return [granuleCumulusId];
 };
+
+/**
+ * Get cumulus IDs for all executions associated to a set of granules
+ *
+ * @param {Knex | Knex.Transaction} knexOrTransaction -
+ *  DB client or transaction
+ * @param {Array<string>} columnNames - column names for whereIn query
+ * @param {Knex.QueryCallback} values - record values for whereIn query
+ * @param {Object} [granulePgModel] - Granule PG model class instance
+ * @param {Object} [granulesExecutionsPgModel]
+ *   Granules/executions PG model class instance
+ * @returns {Promise<number[]>}
+ */
+export const getGranuleExecutionCumulusIds = async (
+  knexOrTransaction: Knex | Knex.Transaction,
+  columnNames: Array<string>,
+  values: Knex.QueryCallback,
```

Comment:
Is `Knex.QueryCallback` the right type here? It seems like that type is for something internal to Knex. In general we're passing these `values` to a `whereIn` clause, so it seems like they have to be an array of any values? So would `Array<any>` be better?

Reply:
Yeah good point. Made some updates to the types

Reply:
This still needs to be addressed